### PR TITLE
Rewriting api

### PR DESCRIPTION
**Pull request by Julianitta**
* added weather api as a separate module
* created core api (AppApi) with additional external (geolocation api, weather api)
* rewritten core api usage in components' needs

[Additional: ]
* added default notification block to weather card, applied styles to it
* restored cloud component rendering

@blobor, pls, take a look

### CODE REVIEW

File: src/core/app.api.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import WeatherApi from './weather-api.js'
+import GeolocationApi from './geolocation-api.js'
+
+const AppApi = {
+  getCityName: () => {
+    return GeolocationApi.getLocationCoords()
+      .then(data => {
```
Comment (blobor) [inline src/core/app.api.js]: Add a little bit magic of object destructuring 😜

File: src/components/weather-input-form.component.jsx [hunk 2]
```diff
@@ -18,14 +18,10 @@ class Form extends React.Component {
...
 
-    let geolocationData = {}
-    const geolocation = WeatherAppApi.getGeolocationCoordinates()
-    geolocation.then(data => {
-      const currentCityName = WeatherAppApi.getCityName(data.currentLatitude, data.currentLongitude)
-      currentCityName.then(data => {
-        this.setState({cityName: `${data}`})
+    AppApi.getCityName()
+      .then(data => {
+        this.setState({cityName: `${data.name}`})
```
Comment (blobor) [inline src/components/weather-input-form.component.jsx]: Object destructuring?

File: src/containers/app.container.jsx [hunk 3]
```diff
@@ -30,7 +30,7 @@ class App extends React.Component {
         <WeatherAppHeader />
         <Form onSubmit={this.showForecastDetails} />
         <WeatherCard card={this.state.infoCard} />
-        {/* <Clouds /> */}
+        <Clouds />
```
Comment (blobor) [inline src/containers/app.container.jsx]: 👍

File: src/components/weather-card.component.jsx [hunk 4]
```diff
@@ -35,10 +32,16 @@ const WeatherCard = ({card}) => {
           </div>
         </div>
         ) : (
+<<<<<<< Updated upstream
```
Comment (blobor) [inline src/components/weather-card.component.jsx]: git conflict?

Comment (Julianitta) [inline src/components/weather-card.component.jsx]: it was a dark side effect of experimenting :scream_cat:

File: src/components/weather-input-form.component.jsx (hunk 2 above)
Comment (Julianitta) [inline src/components/weather-input-form.component.jsx]: @blobor , added
https://github.com/Julianitta/weather-app/pull/5/commits/9c1ddb5c1d9f994d1f4370fcec66627714443167

File: src/components/weather-input-form.component.jsx [hunk 5]
```diff
@@ -20,13 +22,15 @@ class Form extends React.Component {
 
     AppApi.getCityName()
       .then(data => {
-        this.setState({cityName: `${data.name}`})
+        const cityName = data.name
```
Comment (blobor) [inline src/components/weather-input-form.component.jsx]: If you trying to emulate object destructuring, better stop 😄

Comment (blobor) [inline src/components/weather-input-form.component.jsx]: Checkout this https://simonsmith.io/destructuring-objects-as-function-parameters-in-es6/

Comment (Julianitta) [inline src/components/weather-input-form.component.jsx]: :astonished:  fixed already! https://github.com/Julianitta/weather-app/pull/5/commits/5f8f7e8f2b4c1db7fc4f61fbd39c9ebbfaddb5c9

and it was not an emulation :flushed:
it was trials(

File: package.json [hunk 6]
```diff
@@ -71,5 +71,10 @@
       "__tests__"
     ],
     "browser": true
+  },
+  "standard": {
+  "ignore": [
+    "__tests__"
```
Comment (blobor) [inline package.json]: Tests should also match `JS Standard`

Comment (Julianitta) [inline package.json]: @blobor , I think that's the final version of current api. And here was also implementing general cleaning up through linting js/css files.

File: src/core/geolocation-api.js [hunk 7]
```diff
@@ -1,12 +1,22 @@
...
   return new Promise((resolve, reject) => {
+<<<<<<< HEAD
     navigator.geolocation.getCurrentPosition(resolve, reject)
   })
   .then(position => {
     return {
       currentLatitude: position.coords.latitude,
       currentLongitude: position.coords.longitude
     }
+=======
```
Comment (blobor) [inline src/core/geolocation-api.js]: What is that?

Comment (Julianitta) [inline src/core/geolocation-api.js]: hmm, I've seemed to manually fixed that and commited it and before it was merged.
Don't know exact reason, but already fixed that chunk of code.

**Pull request merged**